src/tasks/migration/utils.ts

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import * as fse from "fs-extra";
+import * as path from "path";
+import { debounce } from "lodash";
+import { tasks, window, workspace } from "vscode";
+import { DeprecatedExportJarTaskProvider, ExportJarTaskProvider } from "../../exportJarSteps/ExportJarTaskProvider";
+import { contextManager } from "../../contextManager";
+import { Context } from "../../constants";
+import { DiagnosticProvider } from "./DiagnosticProvider";
+import { CodeActionProvider } from "./CodeActionProvider";
+
+export async function getTasksJsonPaths(): Promise<string[]> {
+    const tasksJsonPaths = [];
+    if (workspace.workspaceFolders?.length) {
+        for (const folder of workspace.workspaceFolders) {
+            const tasksJsonPath = path.join(folder.uri.fsPath, ".vscode", "tasks.json");
+            if (await fse.pathExists(tasksJsonPath)) {
+                tasksJsonPaths.push(tasksJsonPath);
+            }
+        }
+    }
+    return tasksJsonPaths;
+}
+
+export function asyncDebounce(func: any, wait: any, bind: any) {
+    const debounced = debounce(async (resolve, reject, bindSelf, args) => {
+        try {
+            const result = await func.bind(bindSelf)(...args);
+            resolve(result);
+        } catch (error) {
+            reject(error);
+        }
+    }, wait);
+
+    function returnFunc(...args: any[]) {
+        return new Promise((resolve, reject) => {
+            debounced(resolve, reject, bind, args);
+        });
+    }
+
+    return returnFunc;
+}
+
+export async function updateExportTaskType(tasksJsonPath: string): Promise<void> {
+    const content = await fse.readFile(tasksJsonPath);
+    const taskJSON = JSON.parse(content.toString());
+    if (taskJSON) {
+        const taskArray = taskJSON.tasks;
+        if (taskArray instanceof Array) {
+            for (const task of taskArray) {
+                if (task.type === DeprecatedExportJarTaskProvider.type) {
+                    task.type = ExportJarTaskProvider.exportJarType;
+                }
+            }
+        }
+        await fse.writeFile(tasksJsonPath, JSON.stringify(taskJSON, null, "\t"));
```

Comment:
We should get the EOL from the document instance.

Or, maybe we can just simply do a string replacement via regex matching.

Reply:
> We should get the EOL & indent from the document instance.

Yeah we should consider them.

> Or, maybe we can just simply do a string replacement via regex matching.

if we use regex matching, how can we handle if there is the same value in a string constant? (indeed, it's a very rare scenario) IMO it better to keep JSON's schema.

Reply:
Maybe you can deal it as a string, replace the part that needs to be changed (the diagnostic contains range information which could be useful), and write it back - then the indent will not be broken.